crates/uv/src/commands/build.rs

```diff
@@ -0,0 +1,383 @@
+    native_tls: bool,
+    cache: &Cache,
+    printer: Printer,
+) -> Result<BuiltDistributions> {
+    // Extract the resolver settings.
+    let ResolverSettingsRef {
+        index_locations,
+        index_strategy,
+        keyring_provider,
+        allow_insecure_host,
+        resolution: _,
+        prerelease: _,
+        config_setting,
+        no_build_isolation,
+        no_build_isolation_package,
+        exclude_newer,
+        link_mode,
+        upgrade: _,
+        build_options,
+        sources,
+    } = settings;
+
+    let client_builder = BaseClientBuilder::default()
+        .connectivity(connectivity)
+        .native_tls(native_tls);
+
+    let src_dir = if let Some(src_dir) = src_dir {
+        Cow::Owned(std::path::absolute(src_dir)?)
+    } else {
+        Cow::Borrowed(&*CWD)
+    };
+
+    let output_dir = if let Some(output_dir) = output_dir {
+        std::path::absolute(output_dir)?
+    } else {
+        src_dir.join("dist")
+    };
+
+    // (1) Explicit request from user
+    let mut interpreter_request = python_request.map(PythonRequest::parse);
+
+    // (2) Request from `.python-version`
+    if interpreter_request.is_none() {
+        interpreter_request = PythonVersionFile::discover(src_dir.as_ref(), no_config, false)
+            .await?
+            .and_then(PythonVersionFile::into_version);
+    }
+
+    // (3) `Requires-Python` in `pyproject.toml`
+    if interpreter_request.is_none() {
+        let project =
+            match VirtualProject::discover(src_dir.as_ref(), &DiscoveryOptions::default()).await {
+                Ok(project) => Some(project),
+                Err(WorkspaceError::MissingProject(_)) => None,
+                Err(WorkspaceError::MissingPyprojectToml) => None,
+                Err(WorkspaceError::NonWorkspace(_)) => None,
+                Err(err) => {
+                    warn_user_once!("{err}");
+                    None
+                }
+            };
+
+        if let Some(project) = project {
+            interpreter_request = find_requires_python(project.workspace())?
+                .as_ref()
+                .map(RequiresPython::specifiers)
+                .map(|specifiers| {
+                    PythonRequest::Version(VersionRequest::Range(specifiers.clone()))
+                });
+        }
+    }
+
+    // Locate the Python interpreter to use in the environment.
+    let interpreter = PythonInstallation::find_or_download(
+        interpreter_request.as_ref(),
+        EnvironmentPreference::Any,
+        python_preference,
+        python_downloads,
+        &client_builder,
+        cache,
+        Some(&PythonDownloadReporter::single(printer)),
+    )
+    .await?
+    .into_interpreter();
+
+    // Add all authenticated sources to the cache.
+    for url in index_locations.urls() {
+        store_credentials_from_url(url);
+    }
+
+    // Initialize the registry client.
+    let client = RegistryClientBuilder::new(cache.clone())
```

Comment:
Fyi there is a `impl<'a> TryFrom<BaseClientBuilder<'a>> for RegistryClientBuilder<'a>` so we can avoid some repetition here? Kind of annoying it creates a temporary cache though.